airbyte-cdk/python/airbyte_cdk/entrypoint.py

```diff
@@ -21,13 +25,24 @@
 from airbyte_cdk.sources.utils.schema_helpers import check_config_against_spec_or_exit, split_config
 from airbyte_cdk.utils.airbyte_secrets_utils import get_secrets, update_secrets
 from airbyte_cdk.utils.traced_exception import AirbyteTracedException
+from requests import Session
 
 logger = init_logger("airbyte")
 
+VALID_URL_SCHEMES = ["https"]
+CLOUD_DEPLOYMENT_MODE = "cloud"
+
 
 class AirbyteEntrypoint(object):
     def __init__(self, source: Source):
         init_uncaught_exception_handler(logger)
+
+        # DEPLOYMENT_MODE is read when instantiating the entrypoint because it is the common path shared by syncs and connector
+        # builder test requests
+        deployment_mode = os.environ.get("DEPLOYMENT_MODE", "")
+        if deployment_mode.casefold() == CLOUD_DEPLOYMENT_MODE:
+            _init_internal_request_filter()
```

Comment:
We had discussed it, but the reason why we ended up settling on the `DEPLOYMENT_MODE` at least on this iteration is because we know that the environment variable is always set. A new variable would mean we have to make sure is properly assigned across all our infra and is slightly harder to move or adjust later. So I think as a first pass this would be the easiest and doesn't force us down a path.